File: Sample/BasicSample/sample_cpp/File/File_StreamFile.cpp

#include <Altseed.h>

// StreamFile を用いて、ファイルからテキストを読み込むサンプルです。
void File_StreamFile()
{
	// Altseedを初期化する。
	asd::Engine::Initialize(asd::ToAString("File_StreamFile").c_str(), 640, 480, asd::EngineOption());

	// フォントを生成する。
	auto font = asd::Engine::GetGraphics()->CreateFont(asd::ToAString("Data/Font/Font1.aff").c_str());

	// オブジェクトを生成する。
	auto obj = std::make_shared<asd::TextObject2D>();

	// 描画に使うフォントを設定する
	obj->SetFont(font);

	// 描画位置を指定する
	obj->SetPosition(asd::Vector2DF(100, 100));

	// ファイルオブジェクト作成
	auto streamFile = asd::Engine::GetFile()->CreateStreamFile(asd::ToAString("Data/Text/HelloWorld.txt").c_str());

	// ファイルサイズ取得
	const auto size = streamFile->GetSize();

	// ファイルの内容をバッファへ格納
	std::vector<uint8_t> buffer;
	streamFile->Read(buffer, size);

	// 描画したいテキストをファイルから読み取る
	char text[32] = { 0 };
	memcpy(text, buffer.data(), size);

	// 描画する文字列の指定
	obj->SetText(asd::ToAString(text).c_str());

	// オブジェクトのインスタンスをエンジンへ追加する。
	asd::Engine::AddObject2D(obj);

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// Altseedを更新する。
		asd::Engine::Update();
	}

	// Altseedを終了する。
	asd::Engine::Terminate();
}

File: Sample/BasicSample/sample_cs/File/File_StreamFile.cs
using System.Collections.Generic;
using System.Text;

class File_StreamFile : ISample
{
	public string Description
	{
		get { return ""; }
	}
	public string Title
	{
		get { return ""; }
	}
	public string ClassName
	{
		get { return "File_StreamFile"; }
	}

	public void Run()
	{
		// Altseedを初期化する。
		asd.Engine.Initialize("File_StreamFile", 640, 480, new asd.EngineOption());

		// フォントを生成する。
		var font = asd.Engine.Graphics.CreateFont("Data/Font/Font1.aff");

		// オブジェクトを生成する。
		var obj = new asd.TextObject2D();

		// 描画に使うフォントを設定する
		obj.Font = font;

		// 描画位置を指定する
		obj.Position = new asd.Vector2DF(100, 100);

		// ファイルオブジェクト作成
		var staticFile = asd.Engine.File.CreateStreamFile("Data/Text/HelloWorld.txt");

		// ファイルの内容をバッファへ格納
		var buffer = new List<byte>();
		staticFile.Read(buffer, 32);

		// UTF-8でバイト列をデコードする
		char[] text = Encoding.UTF8.GetChars(buffer.ToArray());

		// 描画する文字列の指定
		obj.Text = string.Concat(text);

		// オブジェクトのインスタンスをエンジンへ追加する。
		asd.Engine.AddObject2D(obj);

		// Altseedのウインドウが閉じられていないか確認する。
		while (asd.Engine.DoEvents())
		{
			// Altseedを更新する。
			asd.Engine.Update();
			Recorder.TakeScreenShot("File_StreamFile", 30);
		}

		// Altseedを終了する。
		asd.Engine.Terminate();
	}
}
